legate/core/runtime.py

```diff
@@ -711,14 +729,389 @@ def record_partition(self, index_space, functor, index_partition):
         self._index_partitions[key] = index_partition
 
 
+class FusionChecker(object):
+    def __init__(self, ops, contexts, runtime):
```

Comment:
This class registers a set of fusion constraints. It also currently calls the partitioner, but I will changes this so the partitioner is called outside this class (and the strategies are just passed in as args)